app/controllers/admin/configs_controller.rb

```diff
@@ -100,6 +109,17 @@ def bust_relevant_caches
       CacheBuster.bust("/tags/onboarding")
     end
 
+    # Validations
+    def brand_contrast_too_low
+      hex = params[:site_config][:primary_brand_color_hex]
+      hex.present? && WCAGColorContrast.ratio(hex.delete("#"), "ffffff") < 4.5 # a11y
```

Comment:
I've mentioned this before, but directly sprinkling dependencies into the codebase is often not great (I know this is only used in one place for now). It's often better to create a wrapper object and use that, so if  `WCAGColorContrast` really disappears/goes stale/etc we only have to update the wrapper, not every single call-site. This way you can also encapsulate some of the logic a bit better.

```ruby
class Color
  MIN_CONTRAST = 4.5

  def self.low_contrast?(color, base_color = "ffffff")
    return unless(color.present?)

    WCAGColorContrast.ratio(color.delete("#"), base_color.delete("#")) < MIN_CONTRAST
   end
end
```

Then this just becomes

```ruby
      hex = params[:site_config][:primary_brand_color_hex]
      Color.low_contrast?(hex)
```

which is really all the reader of this validation needs to know 😃

Reply:
Great call. I added a new `Color::Accessibility` class, and it's definitely cleaner.